### ISSUE TITLE
different treatment of train and valid loss?

### ISSUE THREAD

**Issue by tombosc** [pytorch/fairseq#785]
Hello ! :-) 

There's a discrepancy between how the stats are computed for printing in `train()` and `validate()`:

[In the training loop, there is](https://github.com/pytorch/fairseq/blob/master/train.py#L130)
```
            if 'loss' in k:
                extra_meters[k].update(v, log_output['sample_size'])
```

while these lines are not present in validation. Don't think it is very safe to rely on the name containing 'loss' anyways?

Thanks!

**Comment by myleott** [comment]
Hi! The main loss fields don’t have this weighting (see a few lines up), this logic was only added to support secondary losses, but I agree it’s messy. I’m working on a refactoring of logging in fairseq that will get rid of these hacks entirely.

**Comment by benjamin3344** [comment]
> Hi! The main loss fields don’t have this weighting (see a few lines up), this logic was only added to support secondary losses, but I agree it’s messy. I’m working on a refactoring of logging in fairseq that will get rid of these hacks entirely.

Hi, could you explain why the validation loss is larger than the training loss.. I am curious about how they are calculated in a different way.
